homeassistant/components/switchbot/switch.py

```diff
@@ -6,39 +6,56 @@
 import switchbot
 import voluptuous as vol
 
-from homeassistant.components.switch import PLATFORM_SCHEMA, SwitchDevice
-from homeassistant.const import CONF_MAC, CONF_NAME
+from homeassistant.components.switch import (
+    ENTITY_ID_FORMAT,
+    PLATFORM_SCHEMA,
+    SwitchDevice,
+)
+from homeassistant.const import CONF_FRIENDLY_NAME, CONF_MAC, CONF_SWITCHES
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.restore_state import RestoreEntity
 
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_NAME = "Switchbot"
+SWITCH_SCHEMA = vol.Schema(
+    {vol.Required(CONF_MAC): cv.string, vol.Optional(CONF_FRIENDLY_NAME): cv.string}
+)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_MAC): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    }
+    {vol.Required(CONF_SWITCHES): cv.schema_with_slug_keys(SWITCH_SCHEMA)}
 )
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Perform the setup for Switchbot devices."""
-    name = config.get(CONF_NAME)
-    mac_addr = config[CONF_MAC]
-    add_entities([SwitchBot(mac_addr, name)])
+    devices = config.get(CONF_SWITCHES, {})
+    switches = []
+
+    for object_id, device_config in devices.items():
+        switches.append(
+            SwitchBot(
+                object_id,
```

Comment:
Why not still call it `mac_addr` ?

Reply:
I am not using the mac address in this case. Instead I wanted to use an identifier, like in this [example](https://www.home-assistant.io/integrations/switch.command_line/#identifier).

Reply:
But the mac address should be enough. You do not need the identifier

Reply:
Sure I can use the mac address instead of this identifier

Reply:
@Danielhiversen however a mac address cannot be used in a schema with slug keys. Or do you prefer a completely different schema?

Reply:
@Danielhiversen rr am I misunderstanding you and this is what you described? 😅
```yaml
switch:
  - platform: switchbot
    switches:
      000a959d6816:
        name: Kitchen
      000a959d6815:
        name: Bedroom
```